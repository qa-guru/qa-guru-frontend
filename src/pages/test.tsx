import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
