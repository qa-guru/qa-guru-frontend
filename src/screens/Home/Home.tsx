import { gql, useQuery } from "@apollo/client";
import { Result, Spin } from "antd";
import { FormattedMessage } from "react-intl";
import { SmileOutlined } from "@ant-design/icons";

// const QUERY = gql`
//   query {
//     userInfo {
//       username
//     }
//   }
// `;

export const Home = () => {
  // const { loading, error, data } = useQuery(QUERY);

  // if (loading) {
  //   return <Spin />;
  // }

  // if (error != null) {
  //   return (
  //     <Result status="error" title="Query failed" subTitle={error.message} />
  //   );
  // }

  // const { username } = data?.userInfo;

  return (
    <Result
      icon={<SmileOutlined />}
      title={
        <FormattedMessage
          id="home.welcome"
          values={{ appName: "Qaguru Application" }}
        />
      }
      // subTitle={<FormattedMessage id="home.loggedInAs" values={{ username }} />}
    />
  );
};
