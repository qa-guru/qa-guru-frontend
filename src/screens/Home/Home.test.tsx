import { MemoryRouter } from "react-router-dom";
import { render } from "../../test/utilsTest";
import Home from "./Home";

describe("Home", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
