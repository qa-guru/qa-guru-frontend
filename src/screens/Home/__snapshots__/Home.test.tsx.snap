// Vitest Snapshot v1

exports[`Home > the component is render 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root MuiTypography-h4 css-t0sbn-MuiTypography-root"
  >
    Мои курсы
  </h4>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-1ji7omv-MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style="width: 100px; height: 100px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
